### ISSUE TITLE
[feature] [v4] make $colors overridable

### ISSUE THREAD

**Issue by paulsouche** [ionic-team/ionic#15485]
# Feature Request

**Ionic Info**
I do not use ionic-cli sorry...

**Describe the Feature Request**
Be able to override `$colors`

**Describe Preferred Solution**
The hard way is to make `$colors` `!default` in globals.scss

**Describe Alternatives**
 - re-export colors.scss and avoid import globals.scss to override `$colors`.
 - make `$primary`, `$secondary` ... colors overridable (`!default`) 

**Additional Context**
The purpose is to change primary, secondary, ... colors and be able to add ours

```
.ion-color-myColor {
  //
}
```

Thanks anyway for the great job !

**Comment by paulstelzer** [comment]
$colors is no more available in ionic4, instead Ionic is using CSS variables (https://developer.mozilla.org/en-US/docs/Web/CSS/Using_CSS_variables). So colors are already overwriteable in Ionic 4

**Comment by paulsouche** [comment]
Hi,

> $colors is no more available in ionic4

This is wrong ! $colors are part of the core.scss

So why maintain color attributes on components? If I do

```html
<ion-item color="primary">
```

And I can't override primary color without using css variables, this is useless...

Again, purpose on this feature is to generate custom colors

```
.ion-color-primary {
  //my primary color
}


.ion-color-secondary {
  //my secondary color
}
```

**Comment by paulstelzer** [comment]
Are you really talking about Ionic 4? This is the way you set the colors:

In your` global.scss `put the following:

```
// http://ionicframework.com/docs/theming/
@import "~@ionic/angular/css/core.css";
@import "~@ionic/angular/css/normalize.css";
@import "~@ionic/angular/css/structure.css";
@import "~@ionic/angular/css/typography.css";

@import "~@ionic/angular/css/padding.css";
@import "~@ionic/angular/css/float-elements.css";
@import "~@ionic/angular/css/text-alignment.css";
@import "~@ionic/angular/css/text-transformation.css";
@import "~@ionic/angular/css/flex-utils.css";
```

Afterwards in your theme/variables.scss you set the colors as CSS variables which you can overwrite in components etc:

```
:root {
  /** primary **/
  --ion-color-primary: #488aff;
  --ion-color-primary-rgb: 72,138,255;
  --ion-color-primary-contrast: #fff;
  --ion-color-primary-contrast-rgb: 255,255,255;
  --ion-color-primary-shade: #3f79e0;
--ion-color-primary-tint: #5a96ff;
...
}
```

**Comment by paulsouche** [comment]
Indeed, I know that, but my goal is to maintain myColor map and add myCustomColor

```sass
@import '~@ionic/core/dist/collection/themes/ionic.globals';

$my-colors: (
  // ionic overrides
  primary: (
    base:     #256ece,
    contrast: #fff,
    shade:    get-color-shade(#256ece),
    tint:     get-color-tint(#256ece)
  ),
  // business
  error: (
    base:     #ff5757,
    contrast: #fff,
    shade:    get-color-shade(#ff5757),
    tint:     get-color-tint(#ff5757)
  )
);

$colors = extend($colors, $my-colors)

@import '~@ionic/core/dist/collection/css/core'; // this create .ion-color-error
@import '~@ionic/core/dist/collection/css/normalize';
@import '~@ionic/core/dist/collection/css/padding';
@import '~@ionic/core/dist/collection/css/structure';
@import '~@ionic/core/dist/collection/css/text-alignment';
@import '~@ionic/core/dist/collection/css/text-transformation';
@import '~@ionic/core/dist/collection/css/typography';
```

And with that I can do

```html
<ion-button color="error">
```

No css variables needed

Hope this is clearer

**Comment by paulstelzer** [comment]
Ah okay, yeah that would be great, but its still working with CSS Variables. Just add

```
.ion-color-gold {
    --ion-color-base: var(--ion-color-gold) !important;
    --ion-color-base-rgb: var(--ion-color-gold-rgb) !important;
    --ion-color-contrast: var(--ion-color-gold-contrast) !important;
    --ion-color-contrast-rgb: var(--ion-color-gold-contrast-rgb) !important;
    --ion-color-shade: var(--ion-color-gold-shade) !important;
    --ion-color-tint: var(--ion-color-gold-tint) !important;
}
```

and then you can use `<ion-button color="gold">`, but I understand your way - that would be a faster way ;) Let us see what the Ionic dev team will develop ;)

**Comment by marticrespi** [comment]
Can you define more colors in variables.scss like

>   // vibrant
  --ion-color-vibrant: #663399;
  --ion-color-vibrant-rgb: 72,138,255;
  --ion-color-vibrant-contrast: #fff;
  --ion-color-vibrant-contrast-rgb: 255,255,255;
  --ion-color-vibrant-shade: #222;
  --ion-color-vibrant-tint: #307bbe;

If I add a new one (like above) I can't use as commented before..

`<ion-button color="vibrant">`

Or this way is not possible? If not, how can I add a new one?

**Comment by paulstelzer** [comment]
@marticrespi Do not forgot to add `.ion-color-virbant` (in your case) in the css / scss file

**Comment by marticrespi** [comment]
Hi @paulstelzer, how you say works perfectly! Thanks a lot!

**Comment by paulstelzer** [comment]
@marticrespi I also created an post on Medium.com how to make it works -> https://medium.com/@paulstelzer/ionic-4-how-to-add-more-colors-and-use-them-as-color-in-buttons-and-more-7175ab4ae4e7

**Comment by paulsouche** [comment]
Fixed since b790073d773595b9bae2a74c075f98377da47c41
